mutiny-core/src/storage.rs

```diff
@@ -234,6 +253,22 @@ pub trait MutinyStorage: Clone + Sized + 'static {
     /// Deletes all data from the storage
     async fn clear() -> Result<(), MutinyError>;
 
+    /// Deletes all data from the storage and removes lock from VSS
+    async fn delete_all(&self) -> Result<(), MutinyError> {
+        Self::clear().await?;
+        // remove lock from VSS if is is enabled
+        if self.vss_client().is_some() {
+            let device = self.get_device_id()?;
+            // set time to 0 to unlock
+            let lock = DeviceLock { time: 0, device };
+            // still update the version so it is written to VSS
+            let time = now().as_secs() as u32;
+            self.set_data(DEVICE_LOCK_KEY, lock, Some(time))?;
```

Comment:
why does `set_data` want the current time?

Reply:
we use the time as the version

Reply:
Isn't that the problem then? 

> if versioned.time <= kv.version

Reply:
here devicelock.time version is 0, but we are still setting the vss version number to 0

Reply:
this is all incredibly confusing. version is time and time is version and one is set to 0 but not the other.